:sparkles: feat: 고객센터 기본 세팅 엔티티/레포지토리 추가

jiyoon0000 opened the pull request:
-Inquiry: 사용자/로컬 크리에이터가 등록하는 문의 (제목, 내용, 유형, 상태, 작성자, 담당자) 

-InquiryAnswer: 관리자가 작성하는 답변

-InquiryTemplate: 관리자가 관리하는 답변 템플릿 (반복 답변용)

-InquiryType: 문의 유형 (예약, 매거진, 리뷰, 프로그램, 결제, 정산, 계정, 정책, 시스템, 기타) 

-InquiryStatus: 문의 진행 상태 (대기, 진행중, 완료, 취소)

-각 엔티티별 Repository 인터페이스 생성

jiyoon0000 committed:
:sparkles: feat: 고객센터 기본 세팅 엔티티/레포지토리 추가
-Inquiry: 사용자/로컬 크리에이터가 등록하는 문의 (제목, 내용, 유형, 상태, 작성자, 담당자)
-InquiryFile: 문의에 첨부되는 파일 (스크린샷, 인증서 등 증빙자료)
-InquiryAnswer: 관리자가 작성하는 답변
-InquiryTemplate: 관리자가 관리하는 답변 템플릿 (반복 답변용)
-InquiryType: 문의 유형 (예약, 매거진, 리뷰, 프로그램, 결제, 정산, 계정, 정책, 시스템, 기타)
-InquiryStatus: 문의 진행 상태 (대기, 진행중, 완료, 취소)
-각 엔티티별 Repository 인터페이스 생성

j-hann commented on src/main/java/com/salayo/locallifebackend/domain/support/entity/Inquiry.java:
```diff
@@ -0,0 +1,75 @@
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Member assignee;
+
+    @OneToMany(mappedBy = "inquiry", cascade = CascadeType.ALL, orphanRemoval = true)
```
지윤님! 저희가 기본 soft delete로 진행하기로 했던 것으로 기억해요! 문의 entity에 삭제 상태 필드 추가랑 orphanRemoval = true 옵션은 false나 삭제하는게 좋을 것 같다는 의견입니다!

jiyoon0000 replied:
이 부분은 고객센터 삭제가 아닌 고객센터 첨부파일을 InquiryFile 엔티티로 따로 두고, cascade + orphanRemoval로 문의 삭제 시 파일이 함께 삭제가 되도록 설계한 부분인데요, 
다시 생각해보니 공용으로 사용하고 있는 File/FileMapping에서 충분히 관리할 수 있을 것 같아서 InquiryFile 관련 부분을 제거하고 FileMapping에 연결하는 방식으로 수정하겠습니다!!

그리고 지연님이 말씀하시는 softDelete 관련 부분은 삭제 API를 만들면서 추가하려 했던 부분인데, 말씀 주신 김에 바로 반영해두겠습니다!! 감사합니다~☺️

j-hann reviewed (approved):
굿굿굿👍

jiyoon0000 committed:
Merge branch 'develop' into feature/support

jiyoon0000 committed:
:recycle: refactor: Inquiry soft delete 리팩토링 및 InquiryFile 제거
-Inquiry 엔티티에 deletedAt 컬럼 추가
-softDelete() 메서드에서 deletedStatus + deletedAt 함께 처리하도록 수정
-InquiryFile 및 InquiryFileRepository 제거

sooyeoneo reviewed (approved):
💗💗💗
